lib/services/auth/bloc/auth_cubit.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mynotes/services/auth/bloc/auth_state.dart';
+import 'package:mynotes/services/auth/firebase_auth_provider.dart';
+
+class AuthCubit extends Cubit<AuthState> {
+  AuthCubit(this.authProvider)
+      : super(const AuthStateUninitialized(isLoading: true));
+
+  final FirebaseAuthProvider authProvider;
+
+  void sendEmailVerification() async {
+    await authProvider.sendEmailVerification();
+    emit(state);
+  }
+
+  void shouldRegister() {
+    emit(const AuthStateRegistering(
+      exception: null,
+      isLoading: false,
+    ));
+  }
+
+  void initialize() async {
+    await authProvider.initialize();
+    final user = authProvider.currentUser;
+    if (user == null) {
+      emit(const AuthStateLoggedOut(
+        exception: null,
+        isLoading: false,
+      ));
+    } else if (!user.isEmailVerified) {
+      emit(const AuthStateNeedsVerification(isLoading: false));
+    } else {
+      emit(AuthStateLoggedIn(user: user, isLoading: false));
```

Comment:
sealed class are good, but we have some limitations, introduce a status property and make a union class to handle loading loaded success failure, this is how we are maintaing in mostarlab projects